packages/server/src/environment.js

```diff
@@ -71,7 +71,7 @@ module.exports = {
   DEPLOYMENT_CREDENTIALS_URL: process.env.DEPLOYMENT_CREDENTIALS_URL,
   ALLOW_DEV_AUTOMATIONS: process.env.ALLOW_DEV_AUTOMATIONS,
   DISABLE_THREADING: process.env.DISABLE_THREADING,
-  QUERY_THREAD_TIMEOUT: process.env.QUERY_THREAD_TIMEOUT,
+  QUERY_THREAD_TIMEOUT: parseInt(process.env.QUERY_THREAD_TIMEOUT),
```

Comment:
One thing we might run into here is `undefined` being passed to parse int, resulting in `NaN` as the value for `QUERY_THREAD_TIMEOUT`. This isn't so big of a deal given that `NaN` is falsey, but it might be nice to use a helper function so that we retain `undefined` for the value when it's unset. Something like:

```js
const parseIntSafe(number) = () => {
   if (number) {
      return parseInt(number)
   }
}
```

and use that instead

Reply:
Added